### DISCUSSION TITLE
How to read a file in the repo being renovated? I'd like to use a value from that file as a label for the pull request.

### DISCUSSION THREAD

**Discussion by phaninra** [renovatebot/renovate#27027] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.4.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### What is your question?

I'd like to label the PRs with the team names to improve discoverability. The "team name" is mentioned in the `.gitlab-ci.yml` file in the respective repo. Is there a way to plug in some code which can read that file and use the value in `addLabels` ? 

I looked for options like `postUpgradeTasks` but for before the 'renovation' but couldn't find any.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
postUpgradeTasks is the only place where you can run arbitrary code. Unfortunately I can't think of a way to achieve what you're after

**Reply by phaninra** [reply]
Thanks for the response. I wonder there's a way to do it by splitting the task into two jobs. 1) Run renovate to clone the repos, run a postUpgradeTask to gather the info. 2) Do the actual 'renovation'.  
How can I achieve (1) ? How can I tell renovate to visit & clone all the repos but don't do anything than running postUpgradeTasks, i.e. don't create new MRs or touch existing MRs? (2) Is the config.js resolved once at the start or once for each repo. If it's the former, I guess there's no way to do this.
